xbmc/cores/VideoPlayer/DVDFileInfo.cpp

```diff
@@ -306,6 +445,36 @@ bool CDVDFileInfo::ExtractThumb(const CFileItem& fileItem,
   return bOk;
 }
 
+bool CDVDFileInfo::CanExtract(const CFileItem& fileItem)
```

Comment:
Just moved this big rule from the opening line of `CThumbExtractor::DoWork()` and slightly rearranged it.